components/kyma-environment-broker/internal/process/provisioning/staged_manager.go

```diff
@@ -133,6 +133,15 @@ func (m *StagedManager) Execute(operationID string) (time.Duration, error) {
 			if when > 0 {
 				return when, nil
 			}
+
+			// if last step for the last stage finished wihtout issue, set state to domain.Succeeded and publish event
+			if i+1 == len(m.stages) && j+1 == len(stage.steps) {
+				opeartion := processedOperation
+				opeartion.State = domain.Succeeded
+				m.publisher.Publish(context.TODO(), process.ProvisioningSucceeded{
+					Operation: opeartion,
+				})
+			}
```

Comment:
you can send an event outside of the loop, there is a piece of code, which sets the state Succeeded: https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/process/provisioning/staged_manager.go#L144
so you don't need to check if it is the last step.

Reply:
Thanks  @piotrmiskiewicz  for the advice, has done the change. Please kindly review.